### PR TITLE
Add checks for zero uncertainty and length of values list in data schema

### LINKED ISSUE #7: Check that uncertainties are not all zero
See https://gitlab.com/hepcedar/rivet/issues/69 for more information.

Some HEPData tables have bins where all uncertainties are zero.  This causes problems in fitting applications like [Contur](https://hepcedar.gitlab.io/contur-webpage/).  The [hepdata-validator](https://github.com/HEPData/hepdata-validator) should invalidate these cases.

### PR DESCRIPTION
**Pull request by GraemeWatt**
* Check that uncertainties are not all zero in data schema (closes #7).
* Check that length of 'values' list is consistent in data schema.
* Add tests for all zero uncertainties and inconsistent 'values' list.
* Catch LookupError in 'validate' method of SubmissionFileValidator.
* Rename 'valid_data_with_percent.yaml' file and remove invalid fields.
* Change 'error' to 'uncertainty' in tests since 'error' is ambiguous.
* Add 2nd value for independent variable in 'valid_file.yaml' (& json).
* Correct mistake in README.rst, add print_errors in last usage example.

Signed-off-by: Graeme Watt <graeme.watt@durham.ac.uk>

### CODE REVIEW

**Comment by GraemeWatt** [comment]
@alisonrclarke: I've added new functions `check_for_zero_uncertainty` and `check_length_values` in `data_file_validator.py`, as additional checks after the schema validation.  I had a look at "[Validating With Additional Types](https://python-jsonschema.readthedocs.io/en/stable/validate/#validating-with-additional-types)" in the `jsonschema` documentation, but I'm not sure it's easily applicable.  We also want to allow validation using only `data_schema.json` in third-party software ([example](https://cholmcc.gitlab.io/hepdata/api/hepdata.io.Validator.html#hepdata.io.Validator)), without requiring associated Python code.  I'll update the tests to use `pytest` instead of `unittest` after #10 is merged.  Let me know if you have any comments on this implementation.

**Comment by GraemeWatt** [comment]
@alisonrclarke: I've added new functions `check_for_zero_uncertainty` and `check_length_values` in `data_file_validator.py`, as additional checks after the schema validation.  I had a look at "[Validating With Additional Types](https://python-jsonschema.readthedocs.io/en/stable/validate/#validating-with-additional-types)" in the `jsonschema` documentation, but I'm not sure it's easily applicable.  We also want to allow validation using only `data_schema.json` in third-party software ([example](https://cholmcc.gitlab.io/hepdata/api/hepdata.io.Validator.html#hepdata.io.Validator)), without requiring associated Python code.  I'll update the tests to use `pytest` instead of `unittest` after #10 is merged.  Let me know if you have any comments on this implementation.

File: hepdata_validator/data_file_validator.py [hunk 1]
```diff
@@ -121,6 +121,9 @@ def validate(self, **kwargs):
                 json_validate(data, custom_schema)
             else:
                 json_validate(data, default_data_schema)
+                if self.schema_version != '0.1.0':
```
Comment (alisonrclarke) [inline hepdata_validator/data_file_validator.py]: I realise that in practice at the moment, the schema version will only be 0.1.0 or 1.0.0, but I wonder if we could make this less hard-coded. e.g. check the first part of the version number as an integer.

File: hepdata_validator/data_file_validator.py [hunk 2]
```diff
@@ -143,3 +146,64 @@ def __init__(self, message=''):
...
+                            error_plus = float(error_plus)
+                        except ValueError:
+                            pass
+
+                        if isinstance(error_minus, str):
+                            error_minus = error_minus.replace('%', '')
+                        try:
+                            error_minus = float(error_minus)
+                        except ValueError:
+                            pass
```
Comment (alisonrclarke) [inline hepdata_validator/data_file_validator.py]: Could we extract this into a separate method rather than duplicating code?

**Comment by GraemeWatt** [comment]
Thanks, @alisonrclarke, I've addressed your two comments.  Please take another look before I merge this PR.  Then I think I can make the 0.2.0 release now.

File: hepdata_validator/data_file_validator.py [hunk 3]
```diff
@@ -121,7 +121,8 @@ def validate(self, **kwargs):
                 json_validate(data, custom_schema)
             else:
                 json_validate(data, default_data_schema)
-                if self.schema_version != '0.1.0':
+                major_schema_version = int(self.schema_version.split('.')[0])
```
Comment (alisonrclarke) [inline hepdata_validator/data_file_validator.py]: I was going to suggest wrapping this in a try/except, but actually I think checking for a valid schema version should happen when the validator is initialised (mea culpa)- I'll log a separate issue for that.

Comment (alisonrclarke) [reply hepdata_validator/data_file_validator.py]: I've logged #12 to reflect this.
